lib/Dashboard/RecentPagesWidget.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace OCA\Collectives\Dashboard;
+
+use OCA\Collectives\Model\CollectiveInfo;
+use OCA\Collectives\Model\PageInfo;
+use OCA\Collectives\Service\CollectiveService;
+use OCA\Collectives\Service\PageService;
+use OCP\Dashboard\IReloadableWidget;
+use OCP\Dashboard\Model\WidgetItem;
+use OCP\Dashboard\Model\WidgetItems;
+use OCP\IL10N;
+use OCP\IURLGenerator;
+use OCP\IUserSession;
+
+class RecentPagesWidget implements IReloadableWidget {
+	public const REFRESH_INTERVAL_IN_SECS = 33;
+	public const MAX_ITEMS = 10;
+
+	public function __construct(
+		protected IL10N $l10n,
+		protected IURLGenerator $urlGenerator,
+		protected IUserSession $userSession,
+		protected PageService $pageService,
+		protected CollectiveService $collectiveService,
+	) {}
+
+	public function getItemsV2(string $userId, ?string $since = null, int $limit = 7): WidgetItems {
+		if (!($user = $this->userSession->getUser())) {
+			return new WidgetItems();
+		}
+
+		$collectives = $this->collectiveService->getCollectives($user->getUID());
+		$results  = [];
+		foreach ($collectives as $collective) {
+			// fetch pages from the current collective
+			$id = $collective->getId();
+			$pages = $this->pageService->findAll($id, $user->getUID());
+
+			// sort pages and slice to the maximal necessary amount
+			usort($pages, function (PageInfo $a, PageInfo $b): int {
+				return $b->getTimestamp() - $a->getTimestamp();
+			});
+			$pages = array_slice($pages, 0, self::MAX_ITEMS);
+
+			// prepare result entries
+			foreach ($pages as $page) {
+				$results[] = [
+					'timestamp' => $page->getTimestamp(),
+					'page' => $page,
+					'collective' => $collective
+				];
+			}
+
+			// again sort result and slice to the max amount
+			usort($results, function (array $a, array $b): int {
+				return $b['timestamp'] - $a['timestamp'];
+			});
+			$results = array_slice($results, 0, self::MAX_ITEMS);
+		}
+
+		$items = [];
+		foreach ($results as $result) {
+			/* @var array{timestamp: int, page: PageInfo, collective: CollectiveInfo} $result */
+
+			$pathParts = [$result['collective']->getName()];
+			if ($result['page']->getFilePath() !== '') {
+				$pathParts = array_merge($pathParts, explode('/', $result['page']->getFilePath()));
+			}
+			if ($result['page']->getFileName() !== 'Readme.md') {
+				$pathParts[] = $result['page']->getTitle();
+			}
+
+			$iconData = $result['page']->getEmoji()
+				? $this->getEmojiAvatar($result['page']->getEmoji())
+				: $this->getEmojiAvatar('🗒');
```

Comment:
It would be more consistent to use the page icon here instead of a close (but different) looking emoji. I see that you already tried to do that, so I would trust your assessment for now that it's hard to get this right.